src/multilinear.jl

```diff
@@ -296,8 +334,74 @@ function permutation_matrix(
 ) where {T}
     dims = dims isa Integer ? fill(dims, length(perm)) : dims
     d = prod(dims)
-    id = SA.sparse(one(T)*I, (d, d))
+    id = SA.sparse(one(T) * I, (d, d))
     return permute_systems(id, perm, dims; rows_only = true)
 end
 permutation_matrix(dims, perm) = permutation_matrix(Bool, dims, perm)
 export permutation_matrix
+
+@doc """
+    trace_replace(X::AbstractMatrix, remove::AbstractVector, dims::AbstractVector = _equal_sizes(X))
+
+Takes the partial trace of matrix `X` with subsystem dimensions `dims` and replace the removed subsystems by identity.
+If the argument `dims` is omitted two equally-sized subsystems are assumed.
+""" trace_replace(X::AbstractMatrix, remove::AbstractVector, dims::AbstractVector = _equal_sizes(X))
+
+for (T, limit, wrapper) ∈
+    [(:AbstractMatrix, :dpt, :identity), (:(Hermitian), :j, :(Hermitian)), (:(Symmetric), :j, :(Symmetric))]
+    @eval begin
+        function trace_replace(
+            X::$T,
+            replace::AbstractVector{<:Integer},
+            dims::AbstractVector{<:Integer} = _equal_sizes(X)
+        )
+            isempty(replace) && return X
+            length(replace) == length(dims) && return SA.sparse(tr(X) / prod(dims[replace]) * I, size(X))
```

Comment:
```suggestion
            length(replace) == length(dims) && return $wrapper(Matrix(I * tr(X) / size(X, 1), size(X)))
```
otherwise the function is type-unstable